### PR TITLE
test(messaging): add additional messaging test coverage

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

Further unit test coverage for `firebase_messaging_web`.

## Related Issues

none.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by anastaciodevin24 — APPROVED** [review]

**Review by Duc56789 — APPROVED** [review]

**Review by Ehesp — APPROVED** [review]

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 1]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
...
-    test('new instance', () {
-      const testTitle = 'test notification';
-      const notification = RemoteNotification(title: testTitle);
-      expect(notification.title, equals(testTitle));
+  Map<String, dynamic>? mockNotificationMap;
+  Map<String, dynamic>? mockNullNotificationMap;
+
+  group('RemoteNotification', () {
+    setUp(() {
+      mockNotificationMap = {
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:14]:
```suggestion
      mockNotificationMap = const {
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 2]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
...
+        'titleLocArgs': ['titleLocArgs'],
+        'titleLocKey': 'titleLocKey',
+        'body': 'body',
+        'bodyLocArgs': ['bodyLocArgs'],
+        'bodyLocKey': 'bodyLocKey',
+        'android': {},
+        'apple': {},
+      };
+
+      mockNullNotificationMap = {
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:25]:
```suggestion
      mockNullNotificationMap = const {
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart (hunk 1 above)
Comment (rrousselGit) [reply packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:14]: Ah wait no ignore me

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 3]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
...
+        'title': null,
+        'titleLocKey': null,
+        'body': null,
+        'bodyLocKey': null,
+        'android': null,
+        'apple': null,
+      };
+    });
+    test('"RemoteNotification.fromMap" with every possible property expected',
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:34]:
```suggestion

    test('"RemoteNotification.fromMap" with every possible property expected',
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 4]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
 import 'package:flutter_test/flutter_test.dart';
 
 void main() {
-  group('Notification', () {
-    test('new instance', () {
-      const testTitle = 'test notification';
-      const notification = RemoteNotification(title: testTitle);
-      expect(notification.title, equals(testTitle));
+  Map<String, dynamic>? mockNotificationMap;
+  Map<String, dynamic>? mockNullNotificationMap;
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:10]:
```suggestion
  late Map<String, dynamic> mockNotificationMap;
  late Map<String, dynamic> mockNullNotificationMap;
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 5]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
...
+        () {
+      RemoteNotification notification = RemoteNotification(
+          android: const AndroidNotification(),
+          apple: const AppleNotification(),
+          title: mockNotificationMap!['title'],
+          titleLocArgs: mockNotificationMap!['titleLocArgs'],
+          titleLocKey: mockNotificationMap!['titleLocKey'],
+          body: mockNotificationMap!['body'],
+          bodyLocArgs: mockNotificationMap!['bodyLocArgs'],
+          bodyLocKey: mockNotificationMap!['bodyLocKey']);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:76]:
```suggestion
          bodyLocKey: mockNotificationMap!['bodyLocKey'],);
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 6]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
...
+          clickAction: mockAndroidNotificationMap['clickAction'],
+          color: mockAndroidNotificationMap['color'],
+          count: mockAndroidNotificationMap['count'],
+          imageUrl: mockAndroidNotificationMap['imageUrl'],
+          link: mockAndroidNotificationMap['link'],
+          priority: mockAndroidNotificationMap['priority'],
+          smallIcon: mockAndroidNotificationMap['smallIcon'],
+          sound: mockAndroidNotificationMap['sound'],
+          ticker: mockAndroidNotificationMap['ticker'],
+          visibility: mockAndroidNotificationMap['visibility']);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:132]:
```suggestion
          visibility: mockAndroidNotificationMap['visibility'],);
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart [hunk 7]
```diff
@@ -6,37 +6,226 @@ import 'package:firebase_messaging_platform_interface/firebase_messaging_platfor
...
+
+      expect(notification.channelId, mockAndroidNotificationMap['channelId']);
+      expect(
+          notification.clickAction, mockAndroidNotificationMap['clickAction']);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:136]:
```suggestion
          notification.clickAction, mockAndroidNotificationMap['clickAction'],);
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/remote_message_test.dart [hunk 8]
```diff
@@ -4,44 +4,157 @@ import 'package:flutter_test/flutter_test.dart';
 void main() {
   Map<String, dynamic>? mockMessageMap;
 
-  group('RemoteMessage: handle optional fields under sound null safety:', () {
+  Map<String, dynamic>? mockNullableMessageMap;
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/remote_message_test.dart:7]:
```suggestion
  late Map<String, dynamic> mockNullableMessageMap;

```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/remote_message_test.dart [hunk 9]
```diff
@@ -4,44 +4,157 @@ import 'package:flutter_test/flutter_test.dart';
...
+      expect(message.from, mockMessageMap!['from']);
+      expect(message.messageId, mockMessageMap!['messageId']);
+      expect(message.messageType, mockMessageMap!['messageType']);
+      expect(message.mutableContent, mockMessageMap!['mutableContent']);
+
       expect(message.notification, isA<RemoteNotification>());
+      expect(message.notification!.title,
+          mockMessageMap!['notification']['title']);
+      expect(
+          message.notification!.body, mockMessageMap!['notification']['body']);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/remote_message_test.dart:64]:
```suggestion
      expect(message.notification!.title,
          mockMessageMap!['notification']['title'],);
      expect(
          message.notification!.body, mockMessageMap!['notification']['body'],);
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/remote_message_test.dart [hunk 10]
```diff
@@ -4,44 +4,157 @@ import 'package:flutter_test/flutter_test.dart';
...
+          contentAvailable: mockMessageMap!['contentAvailable'],
+          data: mockMessageMap!['data'],
+          from: mockMessageMap!['from'],
+          messageId: mockMessageMap!['messageId'],
+          messageType: mockMessageMap!['messageType'],
+          mutableContent: mockMessageMap!['mutableContent'],
+          notification: RemoteNotification.fromMap({}),
+          sentTime: date,
+          threadId: mockMessageMap!['threadId'],
+          ttl: mockMessageMap!['ttl']);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/remote_message_test.dart:108]:
```suggestion
          ttl: mockMessageMap!['ttl'],);
```

File: packages/firebase_messaging/firebase_messaging_web/test/firebase_messaging_web_test.dart [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+      NotificationSettings notification = getNotificationSettings('granted');
+
+      expect(notification.authorizationStatus, AuthorizationStatus.authorized);
+      expect(notification.alert, AppleNotificationSetting.notSupported);
+      expect(notification.announcement, AppleNotificationSetting.notSupported);
+      expect(notification.badge, AppleNotificationSetting.notSupported);
+      expect(notification.carPlay, AppleNotificationSetting.notSupported);
+      expect(notification.lockScreen, AppleNotificationSetting.notSupported);
+      expect(notification.notificationCenter,
+          AppleNotificationSetting.notSupported);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_web/test/firebase_messaging_web_test.dart:38]:
```suggestion
          AppleNotificationSetting.notSupported,);
```

File: packages/firebase_messaging/firebase_messaging_web/test/firebase_messaging_web_test.dart [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+          getNotificationSettings('default');
+      NotificationSettings randomNotification =
+          getNotificationSettings('random string');
+
+      expect(
+          deniedNotification.authorizationStatus, AuthorizationStatus.denied);
+      expect(defaultNotification.authorizationStatus,
+          AuthorizationStatus.notDetermined);
+      expect(randomNotification.authorizationStatus,
+          AuthorizationStatus.notDetermined);
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_web/test/firebase_messaging_web_test.dart:54]:
```suggestion
      expect(
          deniedNotification.authorizationStatus, AuthorizationStatus.denied,);
      expect(defaultNotification.authorizationStatus,
          AuthorizationStatus.notDetermined,);
      expect(randomNotification.authorizationStatus,
          AuthorizationStatus.notDetermined,);
```

File: packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart (hunk 4 above)
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging_platform_interface/test/notification_test.dart:10]: I don't like the idea of sharing those variables across all tests, as they'll likely change over time.

I'd declare those maps directly inside the tests that needs them

**Review by rrousselGit — APPROVED** [review]
LGTM with some minor changes

**Comment by Salakar** [comment]
@googlebot I consent.

**Pull request merged**